src/App.tsx

```diff
@@ -1,156 +1,104 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
 
-const USER_ID = 0;
+const USER_ID = 6419;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Error | null>(null);
+  const [todoSelector, setTodoSelector] = useState<string | null>('All');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((data) => {
+        setTodos(data);
+      })
+      .catch(() => {
+        setError(new Error("Can't get todos from server!"));
+        setTimeout(() => {
+          setError(null);
+        }, 3000);
+      });
+  }, []);
+
+  const getVisibleTodos = () => {
+    return todos.filter((todo) => {
+      switch (todoSelector) {
+        case 'Active':
+          return !todo.completed;
+        case 'Completed':
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  };
+
+  const visibleTodos = useMemo(getVisibleTodos, [todos, todoSelector]);
```

Comment:
Same error as in previous task. We use useMemo to avoid useless code iteration. Also no need to do filter, if we show all